### ISSUE TITLE
Broken mattn/mruby-require from mruby v2

### ISSUE THREAD

**Issue by ksss** [mruby/mruby#4229]
```
$ cat empty.rb

$ mruby -e 'load "empty.rb"
[1]    14711 segmentation fault  mruby -e

$ lldb mruby
(lldb) target create "./build/test/bin/mruby"
Current executable set to 'mruby' (x86_64).
(lldb) run -e 'load "empty.rb"'                                                                                                    Process 14718 launched: 'mruby' (x86_64)
Process 14718 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x000000010001cf3a mruby`mrb_vm_exec(mrb=0x0000000100302430, proc=0x0000000100801c60, pc="7\x01g") at vm.c:1495
   1492	            syms = irep->syms;
   1493	          }
   1494	        }
-> 1495	        mrb->c->stack[0] = recv;
   1496	        /* pop stackpos */
   1497	        mrb->c->stack = ci->stackent;
   1498	        pc = ci->pc;
```

Implementation of mattn/mruby-require's `load` is

https://github.com/mattn/mruby-require/blob/d43d7b415b2fed06b8301fad719ae731434fef99/src/mrb_require.c#L417-L438

The following patch is necessary to reproduce

```patch
diff --git a/src/mrb_require.c b/src/mrb_require.c
index a14a497..b96b349 100644
--- a/src/mrb_require.c
+++ b/src/mrb_require.c
@@ -93,7 +93,6 @@ realpath(const char *path, char *resolved_path) {
 # define debug(...) ((void)0)
 #endif

-#ifdef USE_MRUBY_OLD_BYTE_CODE
 static void
 mrb_load_fail(mrb_state *mrb, mrb_value path, const char *err)
 {
@@ -107,7 +106,6 @@ mrb_load_fail(mrb_state *mrb, mrb_value path, const char *err)

   mrb_exc_raise(mrb, exc);
 }
-#endif

 static mrb_value
 envpath_to_mrb_ary(mrb_state *mrb, const char *name)
@@ -253,6 +251,11 @@ replace_stop_with_return(mrb_state *mrb, mrb_irep *irep)
     irep->ilen++;
   }
 }
+#else
+static void
+replace_stop_with_return(mrb_state *mrb, mrb_irep *irep)
+{
+}
 #endif

 static void
@@ -289,9 +292,7 @@ load_mrb_file(mrb_state *mrb, mrb_value filepath)
     }
     */

-#ifdef USE_MRUBY_OLD_BYTE_CODE
     replace_stop_with_return(mrb, irep);
-#endif
     proc = mrb_proc_new(mrb, irep);
     MRB_PROC_SET_TARGET_CLASS(proc, mrb->object_class);
```

This problem arises from mruby v2.

Is it a problem of [mattn/mruby-require](https://github.com/mattn/mruby-require) ?

**Comment by matz** [comment]
This issue solved with 98fc887c and the latest `mruby-require`.

**Comment by ksss** [comment]
Thank you @matz and @mattn !
